### PR TITLE
Add swagger gradle file to generate java client api

### PR DESCRIPTION
**Pull request by adazza-daniel**
@ben-adazza

### CODE REVIEW

File: adazza-swagger.gradle [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+classes.dependsOn swaggerClasses
+compileJava.dependsOn compileSwaggerJava
+compileScala.dependsOn compileSwaggerScala
+
+
+dependencies {
+    swaggerCompile "org.springframework.boot:spring-boot-starter-web:1.5.6.RELEASE"
+    swaggerCompile 'io.swagger:swagger-annotations:1.5.16'
+    swaggerCompile 'com.squareup.okhttp:okhttp:2.7.5'
```
Comment (ghost) [inline adazza-swagger.gradle]: Why are these provided here and in runtime scope?

File: adazza-swagger.gradle [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+apply plugin: org.detoeuf.SwaggerCodeGenPlugin
+
+swagger {
+    inputSpec = file('swagger.json')
+    outputDir = file('build/swagger')
+    lang = 'java'
+
+    additionalProperties = [
+            'invokerPackage': 'io.swagger.kylo.client',
+            'modelPackage'  : 'io.swagger.kylo.client.model',
```
Comment (ghost) [inline adazza-swagger.gradle]: You need to figure out a way to have this configured in the consuming cod.e

File: adazza-swagger.gradle (hunk 1 above)
Comment (ghost) [reply adazza-swagger.gradle]: Also these versions need to move into the dep mgmt file.

File: adazza-swagger.gradle [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+
+dependencies {
+    swaggerCompile "org.springframework.boot:spring-boot-starter-web:1.5.6.RELEASE"
+    swaggerCompile 'io.swagger:swagger-annotations:1.5.16'
+    swaggerCompile 'com.squareup.okhttp:okhttp:2.7.5'
+    swaggerCompile 'com.squareup.okhttp:logging-interceptor:2.7.5'
+    swaggerCompile 'com.google.code.gson:gson:2.8.1'
+
+    compile sourceSets.swagger.output
```
Comment (ghost) [inline adazza-swagger.gradle]: This should happen automatically.

File: adazza-swagger.gradle (hunk 2 above)
Comment (adazza-daniel) [reply adazza-swagger.gradle]: When you say consuming code do you mean the consuming gradle file that imports this?

Comment (ghost) [reply adazza-swagger.gradle]: yes.

File: adazza-dependency-management.gradle [hunk 4]
```diff
@@ -44,6 +44,13 @@ allprojects {
       dependency group: "ch.qos.logback", name: "logback-classic", version: "1.0.13"
 
 
+      // Swagger: swagger-codegen java client API dependencies
+      dependency group: "org.springframework.boot", name: "spring-boot-starter-web", version: "1.5.6.RELEASE"
```
Comment (ben-adazza) [inline adazza-dependency-management.gradle]: Move all these  version to dep-mgmt.

File: adazza-swagger.gradle [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    }
+    dependencies {
+        classpath "gradle.plugin.org.detoeuf:swagger-codegen-plugin:1.7.4"
+    }
+}
+
+apply plugin: org.detoeuf.SwaggerCodeGenPlugin
+
+configurations {
+    swagger
```
Comment (ghost) [inline adazza-swagger.gradle]: What does this accomplish?

**Comment by adazza-daniel** [comment]
@ben-adazza updated the PR addressing your comments. Many of lines of code in which you commented on are now removed, moved lib deps in dep-mgmt, reduced lib deps.

File: adazza-swagger.gradle [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+        classpath "gradle.plugin.org.detoeuf:swagger-codegen-plugin:1.7.4"
+    }
+}
+
+apply plugin: org.detoeuf.SwaggerCodeGenPlugin
+
+sourceSets {
+    main {
+        java {
+            srcDir file("${project.buildDir.path}/swagger/src/main/java")
```
Comment (ghost) [inline adazza-swagger.gradle]: for this to look like scrooge it would be:
`buildDir/generated-sources/` the src/main/java is superfluous.

**Pull request closed without merging**